### PR TITLE
fix: [PROD-12924] redirect on workspaces selector when user has no ac…

### PR DESCRIPTION
**Pull request by nborde-CSM**
…cess right to the solution

Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

File: src/components/UserStatusGate/UserStatusGate.js [hunk 1]
```diff
@@ -27,6 +32,12 @@ export const UserStatusGate = ({ children }) => {
     ) : (
       <Navigate to="/accessDenied" replace />
     )
+  ) : solutionUnauthorized ? (
+    location.pathname === '/workspaces' ? (
+      <Workspaces />
+    ) : (
+      <Navigate to="/workspaces" replace />
+    )
```
Comment (csm-thu) [inline src/components/UserStatusGate/UserStatusGate.js:40]: Nested ternary operators gets really difficult to read, can we try to replace them by standard `if`blocks ? (early `return` statements in these `if` blocks will prevent nested conditions)

File: src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js [hunk 2]
```diff
@@ -7,24 +7,41 @@ import { STATUSES } from '../../../commons/Constants';
...
+      type: SOLUTION_ACTIONS_KEY.SET_CURRENT_SOLUTION,
+      status: STATUSES.SUCCESS,
+      solution: data,
+    });
+  } catch (error) {
+    console.error(error);
+    yield put(
+      dispatchSetApplicationErrorMessage(
+        error,
+        t('commoncomponents.banner.', "You don't have permission to access this solution.")
```
Comment (csm-thu) [inline src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js:36]: Lack of permissions is probably not the only error that can be caught here (for example, if the workspace configuration has a typo in the solution id, the webapp may try to fetch a solution that doesn't exist).
Can we have a simple distinction between non-existent solutions and permissions errors (based on the return code, for instance), with a default error for other cases: `something went wrong when fetching the solution "<solution id>"`

Comment (csm-thu) [inline src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js:36]: The translated key is invalid: `commoncomponents.banner.`
Also, we should probably add the solution id in the error message

Comment (nborde-CSM) [reply src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js:36]: It was a temporary empty key, while fixing the main problem.

Comment (nborde-CSM) [reply src/state/sagas/solution/FindSolutionById/FindSolutionByIdData.js:36]: It was a temporary empty key, while fixing the main problem.

File: src/components/UserStatusGate/UserStatusGate.js (hunk 1 above)
Comment (nborde-CSM) [reply src/components/UserStatusGate/UserStatusGate.js:40]: I did a mix of the two forms

File: cypress/commons/utils/apiUtils.js [hunk 3]
```diff
@@ -424,8 +424,12 @@ const interceptGetSolution = (solutionId) => {
   const alias = forgeAlias('reqGetSolution');
   cy.intercept({ method: 'GET', url: interceptionURL, times: 1 }, (req) => {
     if (!stub.isEnabledFor('GET_SOLUTIONS')) return;
+    // if (stub.solutionError != null) {
+    //  req.reply(stub.solutionError);
+    // } else {
     const solutionIdToGet = solutionId ?? req.url.match(interceptionURL)?.[1];
     req.reply(stub.getSolutionById(solutionIdToGet));
+    // }
```
Comment (csm-thu) [inline cypress/commons/utils/apiUtils.js:432]: Remaining commented code to remove

File: cypress/e2e/brewery/SolutionUnauthorized.cy.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+describe('User has no access to the solution', () => {
+  stub.solutionError = {
+    statusCode: 403,
+    body: {
+      title: 'Forbidden',
+      status: 403,
+      detail: `RBAC SOL-unauthorized - User does not have permission read`,
+    },
+  };
+
```
Comment (csm-thu) [inline cypress/e2e/brewery/SolutionUnauthorized.cy.js:19]: There's no need to modify the `stub` object anymore, we can create a `const` to store the object

File: cypress/e2e/brewery/SolutionUnauthorized.cy.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  before(() => {
+    stub.start();
+  });
+
+  after(() => {
+    stub.stop();
+  });
+
+  const customSelectWorkspace = (workspaceId) => {
+    const queries = [apiUtils.interceptPowerBIAzureFunction(), apiUtils.interceptGetScenarios()];
```
Comment (csm-thu) [inline cypress/e2e/brewery/SolutionUnauthorized.cy.js:29]: Maybe we should add a comment here to document that we removed the solution interception from the "usual" queries, and it is replaced by a custom solution interception in the test body

File: cypress/commons/utils/apiUtils.js (hunk 3 above)
Comment (nborde-CSM) [reply cypress/commons/utils/apiUtils.js]: Outdated

**Review by csm-thu — APPROVED** [review]
Some changes suggested, but LGTM :+1:

**Review by esasova — APPROVED** [review]
just one little typo, LGTM :+1:

**Pull request merged**